stats/serializers.py

```diff
@@ -7,7 +7,12 @@ class CartItemListCreateSerializer(serializers.ModelSerializer):
     class Meta:
         model = CartItem
         fields = ["id", "customer", "product", "quantity"]
-        read_only_fields = ["id"]
+        read_only_fields = ["id", "customer"]
+
+    def create(self, validated_data):
+        validated_data["customer"] = self.context["request"].user.customer
+        obj = CartItem.objects.create(**validated_data)
+        return obj
 
```

Comment:
Try as much as you can not to change any data in `create()`, Instead, override the customer field and make it a `SerializerMethodField`

See: https://github.com/abdlrhman08/Distributed-project-backend/blob/0168aabe2dedfa95bae83924f38be3be363e5839/stats/serializers.py#L10-L35